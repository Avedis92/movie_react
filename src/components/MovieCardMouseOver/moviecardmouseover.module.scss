@import "../sasscommonmixxins/flex.scss";
@import "../sasscommonmixxins/button.scss";
@import "../sasscommonmixxins/mediaqueries.scss";

.movieDetailsContainer {
  height: 100%;
  .movieDetails {
    height: 100%;
    background: rgba(0, 0, 0, 0.521);
    padding-left: 5%;
    @include flexElements(flex, column, none, space-evenly);
    .closingButton {
      border-radius: 50%;
      height: 4%;
      width: 8%;
      text-align: center;

      p {
        color: white;
        border-radius: 50%;
        height: 100%;
        font-size: 18px;
        font-weight: bold;
        cursor: pointer;
        transition: 500ms;
        margin: auto;

        &:hover {
          transform: scale(110%);
          background-color: red;
        }
      }
    }
    .titleAndVoteAverage {
      @include flexElements(flex, row, center, space-between);
      .titleAndGenre {
        h2 {
          color: white;
          font-weight: bold;
        }
        .genre {
          @include flexElements(flex, row, center, space-between);
        }
      }
      .vote {
        border: 2px solid rgb(27, 186, 235);
        width: 15%;
        margin-right: 4%;
        border-radius: 5px;
        font-weight: bold;
        text-align: center;
        p {
          color: rgb(27, 186, 235);
          margin-top: 2%;
          margin-bottom: 2%;
          font-size: 20px;
        }
      }
    }
    p {
      color: white;
    }
    .buttonSection {
      height: 8%;
      @include flexElements(flex, row, none, space-evenly);
    }
  }
  @include mobile {
    height: 450px;
  }
}
