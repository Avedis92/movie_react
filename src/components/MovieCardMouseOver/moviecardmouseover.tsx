import { useState } from 'react';
import { useHistory } from 'react-router-dom';
import ModalVideo from 'react-modal-video';
import InfoButtons from '../InfoButtons/infobuttons';
import TrailerPlayer from '../TrailerPlayer/trailerplayer';
import 'react-modal-video/scss/modal-video.scss';
import styles from './moviecardmouseover.module.scss';

interface CardInfo extends Omit<Card, 'genreIds'>{
  handleClicks:()=>void
}

function MovieCardMouseOver({
  id, title, voteAverage, overview, posterPath, handleClicks,
}:CardInfo) {
  const history = useHistory();
  const [isOpen, setIsOpen] = useState(false);
  const [trailerKey, setTrailerKey] = useState('');
  const {
    movieDetailsContainer, movieDetails, closingButton, titleAndVoteAverage, titleAndGenre, vote, buttonSection,
  } = styles;
  return (
    <div className={movieDetailsContainer} style={{ backgroundImage: `url(${posterPath})` }}>
      {isOpen ? (
        <ModalVideo
          channel="youtube"
          isOpen={isOpen}
          videoId={trailerKey}
          onClose={() => setIsOpen(false)}
        />
      )
        : null}
      <div className={movieDetails}>
        <div className={closingButton}>
          <p onClick={() => handleClicks()}>X</p>
        </div>
        <div className={titleAndVoteAverage}>
          <div className={titleAndGenre}>
            <h2>
              {title.length > 16 ? `${title.slice(0, 17)}...` : title}
            </h2>
          </div>
          <div className={vote}>
            <p>{voteAverage}</p>
          </div>
        </div>
        <p>{overview}</p>
        <div className={buttonSection}>
          <TrailerPlayer
            movieId={id}
            buttonType="watchNowButton"
            openMovieTrailers={(key) => {
              setTrailerKey(key);
              setIsOpen(true);
            }}
          />
          <InfoButtons
            moreInfo={() => history.push(`/moviedetailpage?movieId=${id}`)}
          />
        </div>
      </div>
    </div>
  );
}

export default MovieCardMouseOver;
