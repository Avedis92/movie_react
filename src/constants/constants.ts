export const apiKey = '0600befd930f4be78f3d1123f393affc';
export const baseUrl = 'https://api.themoviedb.org/3/';
export const trendingMovieUrl = `${baseUrl}trending/movie/week?api_key=${apiKey}&page=`;
export const topRatedMovieUrl = `${baseUrl}movie/top_rated?api_key=${apiKey}&page=`;
export const upcomingMovieUrl = `${baseUrl}movie/upcoming?api_key=${apiKey}&page=`;
export const movieGenreListUrl = `${baseUrl}genre/movie/list?api_key=${apiKey}`;
export const searchMovieUrl = `${baseUrl}search/movie?api_key=${apiKey}&query=`;
export const postPathUrl = 'https://image.tmdb.org/t/p/w500';
export const defaultBackgroundImage = 'https://media-assets-05.thedrum.com/cache/images/thedrum-prod/s3-news-tmp-10557-netflixa--default--1280.jpg';
export const pageNotFoundBackgroundImage = 'https://cdn3.vectorstock.com/i/1000x1000/67/12/404-error-page-background-vector-20596712.jpg';
export const movieVideoUrl = `${baseUrl}/movie/`;
export const searchMovieByGenreUrl = `${baseUrl}discover/movie?api_key=${apiKey}&with_genres=`;
export const defaultPageNumber = 1;
export const defaultMovieType = 'trendings';
