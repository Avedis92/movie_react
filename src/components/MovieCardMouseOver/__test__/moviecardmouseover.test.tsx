import { render } from '@testing-library/react';
import MovieCardMouseOver from '../moviecardmouseover';
import { postPathUrl } from '../../../constants/constants';

test('Create a snapshot test', () => {
  const {asFragment} = render(<MovieCardMouseOver
    id={297761}
    title="Suicide Squad"
    voteAverage={5.91}
    overview="From DC Comics comes the Suicide Squad, an antihero team of incarcerated supervillains who act as deniable assets for the United States government, undertaking high-risk black ops missions in exchange for commuted prison sentences."
    posterPath={`${postPathUrl}/e1mjopzAS2KNsvpbpahQ1a6SkSn.jpg`}
    handleClicks={() => null}
  />);
  expect(asFragment()).toMatchSnapshot();
});
