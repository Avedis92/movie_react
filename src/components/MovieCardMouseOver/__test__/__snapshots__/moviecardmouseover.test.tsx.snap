// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="movieDetailsContainer"
    style="background-image: url(https://image.tmdb.org/t/p/w500/e1mjopzAS2KNsvpbpahQ1a6SkSn.jpg);"
  >
    <div
      class="movieDetails"
    >
      <div
        class="closingButton"
      >
        <p>
          X
        </p>
      </div>
      <div
        class="titleAndVoteAverage"
      >
        <div
          class="titleAndGenre"
        >
          <h2>
            Suicide Squad
          </h2>
        </div>
        <div
          class="vote"
        >
          <p>
            5.91
          </p>
        </div>
      </div>
      <p>
        From DC Comics comes the Suicide Squad, an antihero team of incarcerated supervillains who act as deniable assets for the United States government, undertaking high-risk black ops missions in exchange for commuted prison sentences.
      </p>
      <div
        class="buttonSection"
      >
        <div
          class="watchNowContainer"
        >
          <button
            id="297761"
            name="watchNowButton"
            type="button"
          >
            Watch Now
          </button>
        </div>
        <div
          class="movieInfoButtons"
        >
          <button
            name="moreinfo"
            type="button"
          >
            More Info
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
